boards/stm32f429i-disc1/Makefile.dep

```diff
@@ -9,3 +10,8 @@ endif
 ifneq (,$(filter touch_dev,$(USEMODULE)))
   USEMODULE += stmpe811
 endif
+
+ifneq (,$(filter l3gxxxx,$(USEMODULE)))
+  USEMODULE += i3g4250d
```

Comment:
@benpicco Here we have a problem with different revisions of the board: L3GD20 (Rev. C01, D01), I3G4250D (Rev. E02)

Reply:
Can the sensor type be detected at run-time?

Reply:
In the former implementation it was detected during initialization. At least L3GD20* and [IAL]3G4* have different chip IDs. When I optimized the code for size, I removed this chip detection and defined respective pseudomodules, since the capabilities differ slightly. With the pseudo modules, only the code for the capabilities that the sensor has is compiled in.

For the I2C interface we have additionally the problem that the I2C addresses differ.

Reply:
BTW, the L3GD20 variant isn't active any longer. That's why newer revisions of the board only have I3G4250.

Reply:
Could we select multiple pseudo-modules and only do the detection for the pseudo-modules that are compiled in? If there is only a single one, that should not increase the code size.

Reply:
[Here](https://github.com/RIOT-OS/RIOT/blob/461f9b103e630d5a69162158980f2edd683eb350/drivers/l3gxxxx/l3gxxxx.c#L137) we would have to add an additional check for L3GD20H if pseudomodule `i3g4250d` is together with pseudomodules `l3gd20h` and `l3gxxxx_low_odr` since the I3G4250D doesn't have the register `LOW_ORD` and doesn't support low data rates. The detection itself would add additional code.

Reply:
While trying to change it to detect the sensor at runtime, I realized the following problems:
-  I2C addresses differ for the [variants](https://github.com/RIOT-OS/RIOT/blob/bd270e8e064b793d90a68d07ea288a738ba7048e/drivers/include/l3gxxxx.h#L991), so it is not clear which one should be used as default configuration if pseudomodules `l3gd20*` and `x3g42*` are used simultaneously.
- Full scaling is limited to 245 dps for the A3G4250D. This cannot be reliably checked since its chip ID is the same as for the I3G4250D and L3G4200D.
- `Kconfig` requires the selection of a sensor variant, which in turn defines some dependencies on other configurations like [low ODR availability](https://github.com/RIOT-OS/RIOT/blob/bd270e8e064b793d90a68d07ea288a738ba7048e/drivers/l3gxxxx/Kconfig#L102) or [possible full scales](https://github.com/RIOT-OS/RIOT/blob/bd270e8e064b793d90a68d07ea288a738ba7048e/drivers/l3gxxxx/Kconfig#L172).

Maybe, you have an idea how to deal with such differences. I don't know how A3G4250D behaves if the application tries to set a full scale of 2000 dps.

Reply:
@benpicco I have pushed a commit that includes these changes for testing. If this approach works, I would split it up into separate fixup commits of the driver. What the approach can't solve is that `Kconfig` uses the variant and the corresponding dependencies that is found first.

Reply:
@benpicco have you had time to take a look into this approach?

Reply:
What about adding a board configuration to determine which revision to use? Based on that the correct sensor variant is used. The configuration can be a pseudomodule, having the latest revision as the default one.